### DISCUSSION TITLE
[PROPOSAL] Multi Bumpkin Feature - Tent & Slots

### DISCUSSION THREAD

**Discussion by TheDcol** [sunflower-land/sunflower-land#3134] [Ideas]
Hello, I would like to outlay a proposal for a multi bumpkin feature allowing players to have more wearables active at once.

Players have A LOT OF WEARABLES at the moment.. it is becoming very annoying having to swap all the clothes every x hours for the boosts to apply. I think the community would appreciate a system to simplify this. I propose a way to knock 2 birds with 1 stone via Multi Bumpkins - Tent & Slots:

**Tent Building:** Hosts Bumpkin/s

![image](https://github.com/sunflower-land/sunflower-land/assets/108110065/9756bb95-dc18-4e6f-91f7-d04e7d1fee33)
 
**Procedure:** Via the Tent, deposit 2nd bumpkin from your wallet. This makes the 2nd bumpkin 'Active', opening 4 slots to equip wearables, 2 of which are locked. Selecting a bumpkin in the tent and equipping the slots with a boost wearable makes the item active in the farm.

![image](https://github.com/sunflower-land/sunflower-land/assets/108110065/b26fee24-2568-464e-97bf-e390b794ffd4)

**Options:** Upgrade the Tent to host more than 1 Bumpkin and/or increase slots per bumpkin. Upgrading the tent and unlocking slots would require resources. Potential to add a running cost (having to feed the bumpkins in the tent to keep the slots open?) 

![image](https://github.com/sunflower-land/sunflower-land/assets/108110065/99c5371d-56a0-4ae2-82e9-291456d75bb2)

**Conditions:** Cannot remove item/s from 2nd Bumpkin AND/OR remove bumpkin from the Tent until boosts are on cooldown (similar to current withdrawal rules) to avoid potential exploits. 2nd or 3rd cannot be swapped to main bumpkin within the farm.

**Conclusion:** A short and sweet proposal, this would sort out the Tent AND Multi-Bumpkin mechanic which I think needs to arrive sooner than later, especially with all these cool wearables that we have :D

What do you think? Is it fairly simple to understand? Do you hate it? Open for constructive feedback :)
Cheers

**Comment by vitt0** [comment]
Sounds like a great solution, the idea is just to have the slot linked to the bumpkin like If I equip in the first slot a Eggplant Onesie is going to change how my secondary bumpkin looks too? If soo I think it would be good to have a second tab on the tend that allow us to swap around all the wearables that the secondary bumpkins have but the "tent slot" are only used by wearables that have boost

Atm since bumpkin.io is disable it's really hard to swap the wearables of secondary bumpkins, I imagine this would be a lot harder to do so maybe could be something to add later and the first version of the tent is just "equip" the wearables to the tend but they not actually equipped on the secondary bumpkins

**Reply by TheDcol** [reply]
Thanks VIttoc!

>  I imagine this would be a lot harder to do so maybe could be something to add later and the first version of the tent is just "equip" the wearables to the tend but they not actually equipped on the secondary bumpkins

Yup the first iteration would be to just get the wearable items equipped and working. I have not thought about all the technical aspects yet as far as swapping secondary bumpkins so it's good you bring that up.

Do you think if those secondary bumpkins were just read-only and they just existed outside your tent it would be ok?

I envision a player basically using the bumpkins in the tent as 'mannequins', purely to equip boosts. Your primary bumpkin could also equip boosts but you might dress it up differently and even using non-boost items on it to show off at the plaza (especially if you can get all the boost items on your tent bumpkins)

**Reply by vitt0** [reply]
> Do you think if those secondary bumpkins were just read-only and they just existed outside your tent it would be ok?

I fine with it, I was just thinking it would be good to add the option of swaping secondary bumpkins wearables somewhere, since without bumpkins.io is kinda annoying, but of course I don't think is a priority, if adding this would mean delaying the rest of the idea I wouldn't want that

Just having them as 'mannequins' would be good enought Imo 😄

**Comment by bayselb** [comment]
What if all boosted wearables just took effect once deposited into your farm. Would make bumpkin customisation a lot easier if we didn't have to equip particular wearables for particular boosts

**Reply by TheDcol** [reply]
That is an easy solution, although, I think we should also try to incorporate the Tent and add some costs considering all the extra boosts will become active, IMO

**Comment by Dionis404** [comment]
The idea of multi bambkins is very good, how do you look at what would make the tent multi-level (perhaps an improvement to the house), so you can increase the burning of resources and slightly limit the bonuses from the things carried. 

For example:
A tent allows you to add one more bumpkin to the farm, and 1 thing that gives a bonus
Tier 2 awning, 2 things that give a bonus 
Tier 3 awning 3 things that give a bonus 
House 1 level 2 bumpkins, the first bumpkin can have 3 clothes with bonus the second 1 thing. 
And so on. 

It is also possible to tie it to the level of bumpkin that would burn more food.

**Reply by TheDcol** [reply]
Tent progression/updates would be cool imo! Also adds more resource sinks in the game as players are enabling all the utility wearables

**Comment by adamhannigan** [comment]
Awesome post mate 👏 I will be following this closely for community feedback and ideas - I will wait a little longer for some more community discussion before I hijack the post with a technical deep dive 😅 

From a gameplay perspective, I few areas to keep in mind:

- How intuitive will the Bumpkin + slot allocation feel to a new player. Without a guide, would someone understand the restrictions we put in place.
- How do we balance the progression and reward of unlocking/upgrading the tents/slots. To me, it feels like a wasted opportunity if we don't highlight this as a huge in-game achievement. As such, we need to make sure the effort put in is equally gratifying once reached.

From a technical perspective, we will need to consider exploits/abuse of this feature. Unfortunately the freedom to deposit/withdraw multiple Bumpkins can lead to some risks and unforeseen player exploits. Putting timers, withdraw locks + restrictions often work as bandages and new ways to workaround often appear. Do you have any suggestions on a simple secure solution for managing multiple Bumpkins?

**Reply by TheDcol** [reply]
Thanks Adam,

> Do you have any suggestions on a simple secure solution for managing multiple Bumpkins?

Permanently locking a bumpkin deposited in the tent is 1 solution (albeit, drastic and extreme). Once you deposit a bumpkin in the tent to act as a mannequin, it is forever locked or tied to the farm.

A less extreme solution would be to disable bumpkin swapping within the farm AND place restrictions on withdrawing. Players should not be able to swap their primary and secondary bumpkins within the farm. A player should not be able to withdraw ANY of their bumpkins if ANY of their boosts are currently active.

> How intuitive will the Bumpkin + slot allocation feel to a new player. Without a guide, would someone understand the restrictions we put in place.

A tutorial or 'help' tab within the Tent would help

> How do we balance the progression and reward of unlocking/upgrading the tents/slots. To me, it feels like a wasted opportunity if we don't highlight this as a huge in-game achievement. As such, we need to make sure the effort put in is equally gratifying once reached.

Progression within the tent could be explored:
1- Players need to perform x tasks to unlock x slots/bumpkins to be housed
2- Players need to spend x resources to unlock more bumpkins and slots, or upgrade their tents
3- A combination or rather, option to have both

**Comment by poteitouweb3** [comment]
Hey! Just read all comments and would like to add here my thoughts:

1. As @adamhannigan well previewed the security is the first thing to be considered. None ideia or solution will be useful if we'll need to rollback due to miss sec mapping all risk points.
2. For a fast iteration and feedback i would suggest a functional and basic solution to start: Deposit a 2nd bumpkin > Wear 2~4 SFTs > Enjoy buffs!
3. After a period of tests and feedback we may keep bringing more ideias to reality 🙌🏻

**Comment by Dionis404** [comment]
We must not forget that identical things should not work together. You also need to understand which slot to open first. Here you need to first of all think about new players and at the same time not give a strong boost to the whales.

For safety, you need to use an output timer; this can be used not only for this situation, but in general for everything. All entered SFTs can only be withdrawn after 24 hours. (unless of course there are other restrictions)

**Reply by Dionis404** [reply]
Checked out wearable items with bonuses, and for the first slots, you can go with a **Necklace**, **Onesie**, or **Coat**. In these categories, there are budget-friendly items that will provide good bonuses for beginners without significantly affecting the boosts for "whales"

**Reply by Dionis404** [reply]
It is also essential to understand how bonuses from items on the non-main bumpkin will be distributed outside the farm. For instance, a pirate potion – whether it needs to be on the main bumpkin or if it can be on a secondary one and still provide its bonus (I am aware that it currently does not work).

**Comment by Dionis404** [comment]
![image](https://github.com/sunflower-land/sunflower-land/assets/133494217/dcd5fba4-07c7-4b15-89b8-9b9a25148902)
![image](https://github.com/sunflower-land/sunflower-land/assets/133494217/6744239a-c8b5-4420-a6c3-9c470c7f0a04)
Interface options

**Reply by TheDcol** [reply]
Ohh I like these options!

The one with the bumpkins separated by tabs looks great
